### PR TITLE
SCP-2512: plutus-use-cases-scripts is now run during Hydra CI

### PR DESCRIPTION
**Pull request by koslambrou**
plutus-use-cases-scripts is now run during Hydra CI and the result is available for download from Hydra.

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: plutus-use-cases/default.nix [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+let
+  puc-scripts-invoker = haskell.packages.plutus-use-cases.components.exes.plutus-use-cases-scripts;
+
+  generated-puc-scripts-output = pkgs.runCommand "plutus-use-cases-scripts-output" { } ''
+    mkdir -p $out/scripts
+    mkdir -p $out/transactions
+    ln -s ${haskell.packages.plutus-use-cases.src}/scripts/protocol-parameters.json protocol-parameters.json
+    ${puc-scripts-invoker}/bin/plutus-use-cases-scripts $out/scripts scripts
+    # Mainnet address is used because no networkid is specified (with the '-n' flag)
+    ${puc-scripts-invoker}/bin/plutus-use-cases-scripts $out/transactions transactions -p protocol-parameters.json
```
Comment (j-mueller) [inline plutus-use-cases/default.nix:11]: can you also please make a tar or zip file containing both folders? That's easier than downloading each script individually

Comment (koslambrou) [reply plutus-use-cases/default.nix:11]: Oh yeah, good idea!

**Review by j-mueller — APPROVED** [review]

**Comment by koslambrou** [comment]
@j-mueller Satisfied with the downloadable artifacts? Examples:

https://hydra.iohk.io/build/7010766/download/1/plutus-use-cases-scripts-output/all-outputs.tar.gz
https://hydra.iohk.io/build/7010766/download/1/plutus-use-cases-scripts-output/scripts/auction_1-1.flat
https://hydra.iohk.io/build/7010766/download/1/plutus-use-cases-scripts-output/transactions/auction_1-1.json
https://hydra.iohk.io/build/7010766/download/1/plutus-use-cases-scripts-output/transactions/currency-2.cbor

**Comment by j-mueller** [comment]
Yes that looks good, thanks!

File: plutus-use-cases/default.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{ pkgs, haskell }:
+let
+  puc-scripts-invoker = haskell.packages.plutus-use-cases.components.exes.plutus-use-cases-scripts;
+
+  generated-puc-scripts-output = pkgs.runCommand "plutus-use-cases-scripts-output" { } ''
+    mkdir -p $out/scripts
+    mkdir -p $out/transactions
+    ln -s ${haskell.packages.plutus-use-cases.src}/scripts/protocol-parameters.json protocol-parameters.json
```
Comment (michaelpj) [inline plutus-use-cases/default.nix:8]: You can just pick up the file from the source tree, no need to do this roundabout thing. Something like `${builtins.readFile ./scripts/protocol-paramters.json}`

Comment (michaelpj) [reply plutus-use-cases/default.nix:8]: Not a big deal, though.

**Pull request merged**